src/components/PostCard/PostBodyProposal/PostBodyProposal.js

```diff
@@ -0,0 +1,140 @@
+  proposalStatus,
+  proposalType,
+  proposalOptions,
+  proposalVotes,
+  isAnonymousVote,
+  startTime,
+  quorum,
+  endTime,
+  groups,
+  id
+}) {
+  const dispatch = useDispatch()
+  const { t } = useTranslation()
+  const proposalOptionsArray = (proposalOptions && proposalOptions.items) ? proposalOptions?.items : []
+  const proposalVotesArray = (proposalVotes && proposalVotes.items) ? proposalVotes?.items : []
+
+  const [currentUserVotes, setCurrentUserVotes] = useState(proposalVotesArray.filter(vote => vote?.user?.id === currentUser.id))
+  const [currentUserVotesOptionIds, setCurrentUserVotesOptionIds] = useState(currentUserVotes.map(vote => vote.optionId))
+  const [proposalVoterCount, setProposalVoterCount] = useState(calcNumberOfVoters(proposalVotesArray))
+  const [numberOfPossibleVoters, setNumberOfPossibleVoters] = useState(calcNumberOfPossibleVoters(groups))
+
+  useEffect(() => {
+    setCurrentUserVotes(proposalVotesArray.filter(vote => vote?.user?.id === currentUser.id))
+    setProposalVoterCount(calcNumberOfVoters(proposalVotesArray))
+  }, [proposalVotesArray])
+
+  useEffect(() => {
+    setCurrentUserVotesOptionIds(currentUserVotes.map(vote => vote.optionId))
+  }, [currentUserVotes])
+
+  useEffect(() => {
+    setNumberOfPossibleVoters(calcNumberOfPossibleVoters(groups))
+  }, [groups])
+
+  function handleVote (optionId) {
+    if (proposalType === PROPOSAL_TYPE_SINGLE || proposalType === PROPOSAL_TYPE_MAJORITY) {
+      if (currentUserVotesOptionIds.includes(optionId)) {
+        dispatch(removeProposalVote({ optionId, postId: id }))
+      } else if (currentUserVotesOptionIds.length === 0) {
+        dispatch(addProposalVote({ optionId, postId: id }))
+      } else {
+        const removeOptionId = currentUserVotesOptionIds[0]
+        dispatch(swapProposalVote({ postId: id, addOptionId: optionId, removeOptionId }))
+      }
+    }
+    if (proposalType === PROPOSAL_TYPE_MULTI_UNRESTRICTED) {
+      if (currentUserVotesOptionIds.includes(optionId)) {
+        dispatch(removeProposalVote({ optionId, postId: id }))
+      } else {
+        dispatch(addProposalVote({ optionId, postId: id }))
+      }
+    }
+  }
+
+  const handleVoteThrottled = throttle(200, handleVote)
+
+  return (
+    <div styleName={cx('proposal-body-container', { discussion: proposalStatus === PROPOSAL_STATUS_DISCUSSION, voting: proposalStatus === PROPOSAL_STATUS_VOTING, casual: proposalStatus === PROPOSAL_STATUS_CASUAL, completed: proposalStatus === PROPOSAL_STATUS_COMPLETED })}>
+      <div styleName={cx('proposal-status')}>
+        {proposalStatus === PROPOSAL_STATUS_DISCUSSION && t('Discussion in progress')}
+        {proposalStatus === PROPOSAL_STATUS_VOTING && t('Voting open')}
+        {proposalStatus === PROPOSAL_STATUS_COMPLETED && t('Voting ended')}
+        {proposalStatus === PROPOSAL_STATUS_CASUAL && t('Voting open')}
+      </div>
+      <div styleName={cx('proposal-timing')}>
+        {!startTime && t('Open timeframe')}
```

Comment:
to discuss: this language. Open timeframe is not totally clear to me but I dont immediately have a better idea

Reply:
@cpolitano I think this is one of the copy things Tibet was referring to a few sessions ago